core/arch/arm/kernel/secure_partition.c

```diff
@@ -1418,111 +1418,76 @@ static const struct ts_ops sp_ops = {
 
 static TEE_Result process_sp_pkg(uint64_t sp_pkg_pa, TEE_UUID *sp_uuid)
 {
-	enum teecore_memtypes mtype = MEM_AREA_RAM_SEC;
+	enum teecore_memtypes mtype = MEM_AREA_TA_RAM;
 	struct sp_pkg_header *sp_pkg_hdr = NULL;
-	TEE_Result res = TEE_SUCCESS;
-	tee_mm_entry_t *mm = NULL;
 	struct fip_sp *sp = NULL;
 	uint64_t sp_fdt_end = 0;
 	size_t sp_pkg_size = 0;
 	vaddr_t sp_pkg_va = 0;
-	size_t num_pages = 0;
 
-	/* Map only the first page of the SP package to parse the header */
+	/* Check the first page which contains the SP package header */
 	if (!tee_pbuf_is_sec(sp_pkg_pa, SMALL_PAGE_SIZE))
 		return TEE_ERROR_GENERIC;
 
-	mm = tee_mm_alloc(&tee_mm_sec_ddr, SMALL_PAGE_SIZE);
-	if (!mm)
-		return TEE_ERROR_OUT_OF_MEMORY;
-
-	sp_pkg_va = tee_mm_get_smem(mm);
-
-	if (core_mmu_map_contiguous_pages(sp_pkg_va, sp_pkg_pa, 1, mtype)) {
-		res = TEE_ERROR_GENERIC;
-		goto err;
+	sp_pkg_va = (vaddr_t)phys_to_virt(sp_pkg_pa, mtype, SMALL_PAGE_SIZE);
+	if (!sp_pkg_va) {
+		EMSG("Cannot find mapping for PA %#" PRIxPA, sp_pkg_pa);
+		return TEE_ERROR_GENERIC;
 	}
 
 	sp_pkg_hdr = (struct sp_pkg_header *)sp_pkg_va;
 
 	if (sp_pkg_hdr->magic != SP_PKG_HEADER_MAGIC) {
 		EMSG("Invalid SP package magic");
-		res = TEE_ERROR_BAD_FORMAT;
-		goto err_unmap;
+		return TEE_ERROR_BAD_FORMAT;
 	}
 
 	if (sp_pkg_hdr->version != SP_PKG_HEADER_VERSION_V1 &&
 	    sp_pkg_hdr->version != SP_PKG_HEADER_VERSION_V2) {
 		EMSG("Invalid SP header version");
-		res = TEE_ERROR_BAD_FORMAT;
-		goto err_unmap;
+		return TEE_ERROR_BAD_FORMAT;
 	}
 
 	if (ADD_OVERFLOW(sp_pkg_hdr->img_offset, sp_pkg_hdr->img_size,
 			 &sp_pkg_size)) {
 		EMSG("Invalid SP package size");
-		res = TEE_ERROR_BAD_FORMAT;
-		goto err_unmap;
+		return TEE_ERROR_BAD_FORMAT;
 	}
 
 	if (ADD_OVERFLOW(sp_pkg_hdr->pm_offset, sp_pkg_hdr->pm_size,
 			 &sp_fdt_end) || sp_fdt_end > sp_pkg_hdr->img_offset) {
 		EMSG("Invalid SP manifest size");
-		res = TEE_ERROR_BAD_FORMAT;
-		goto err_unmap;
+		return TEE_ERROR_BAD_FORMAT;
 	}
 
-	core_mmu_unmap_pages(sp_pkg_va, 1);
-	tee_mm_free(mm);
-
-	/* Map the whole package */
+	/* Check the whole SP package now that the size is known */
 	if (!tee_pbuf_is_sec(sp_pkg_pa, sp_pkg_size))
```

Comment:
This is also taken care of implicitly by `phys_to_virt()` below.

Reply:
Fixed